### ISSUE TITLE
Datasource vcd_org_vdc not returning provider_vdc_name

### ISSUE THREAD

**Issue by NilsBusche** [vmware/terraform-provider-vcd#945]
### Terraform Version
```
Terraform v1.3.5
on linux_amd64
+ provider registry.terraform.io/vmware/vcd v3.8.0
```

### Affected Resource(s)
Please list the resources as a list, for example:
- Datasource vcd_org_vdc

### Terraform Configuration Files

```hcl
data "vcd_org_vdc" "org_vdc" {
  org = "org-1234"
  name = "org-vdc-abcd"
}

output "org_vdc_id" {
  value = data.vcd_org_vdc.org_vdc.id
}

output "provider_vdc_name" {
  value = data.vcd_org_vdc.org_vdc.provider_vdc_name
}

output "provider_vdc_name_null" {
  value = data.vcd_org_vdc.org_vdc.provider_vdc_name == null ? "provider_vdc_name is null" : "provider_vdc_name is NOT null"
}
```

### Expected Behavior
- Output `provider_vdc_name` should contain the provider VDC name of the organization VDC
- Debugging output `provider_vdc_name_null` should contain `provider_vdc_name is NOT null`

### Actual Behavior
- Provider VDC name is null in the datasource
- Output `provider_vdc_name` is null
- Debugging output `provider_vdc_name_null` contains `provider_vdc_name is null`
```
data.vcd_org_vdc.org_vdc: Reading...
data.vcd_org_vdc.org_vdc: Read complete after 1s [id=urn:vcloud:vdc:xxxxxxxx]

Changes to Outputs:
  + org_vdc_id             = "urn:vcloud:vdc:xxxxxxxx"
  + provider_vdc_name_null = "provider_vdc_name is null"

You can apply this plan to save these new output values to the Terraform state, without changing any real infrastructure.

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

org_vdc_id = "urn:vcloud:vdc:xxxxxxxx"
provider_vdc_name_null = "provider_vdc_name is null"
```

### Steps to Reproduce
1. `terraform apply`

### User Access rights 
Used user is a user which has only access to one specific tenant, not a system/provider user

### Important Factoids
Using this datasource is required for using the datasource `vcd_vm_placement_policy` and the new attribute `placement_policy_id` in the `vapp_vm` resource as the provider vdc name is a required input for the datasource, e. g.:
```hcl
data "vcd_org_vdc" "org_vdc" {
  org = "org-1234"
  name = "org-vdc-abcd"
}

data "vcd_vm_placement_policy" "policy" {
  name            = "policy-xyz"
  provider_vdc_id = data.vcd_provider_vdc.org_vdc.id
}

resource "vcd_vapp_vm" "vm" {
  [...]

  placement_policy_id = data.vcd_vm_placement_policy.policy.id

  [...]
}
```

Using a system/provider user with limited access rights for only maintaining virtual machines in different organizations also fails because of insuffient rights:
> Error: error retrieving network pool: error fetching network pool: API Error: 403: [ xxxxxxx ] This operation is denied.

The user has all required rights for maintaining VMs, even for assigning them to a placement policy with native VCD API/UI or the former workaround by using `sizing_policy_id`. It seems that there would be additional access rights on other layers required to get the described usecase for placement polices working, which are not needed if you use native VCD API/UI.

**Comment by adambarreiro** [comment]
Hi @NilsBusche, thanks for reporting!

In this case, `provider_vdc_name` is only populated if the user is a system/provider user, as unfortunately Provider VDC information is not accesible for tenant users.

The same limitation goes for the VM Placement Policy, uniqueness of the policy is granted by the Provider VDC where it was configured (in other words, one could have more than one policy with **same name** but that are configured in different Provider VDCs), so we need Provider VDC information to fetch them unequivocally, that's why a system/provider user is needed.

May I suggest you to try to fetch Org VDC + VM Placement Policy with a Terraform provider config using a System administrator and then create the VMs with another Terraform provider config with tenant user? Like this:

```hcl
terraform {
  required_providers {
    vcd = {
      source  = "vmware/vcd"
      version = ">= 3.8.0"
    }
  }
}

provider "vcd" {
  user                 = "administrator"
  password             = "******"
  auth_type            = "integrated"
  url                  = "https://my-vcd.com/api"
  sysorg               = "System"
  org                  = "test"
  allow_unverified_ssl = "true"
  logging              = true
}

provider "vcd" {
  alias                = "tenant"
  user                 = "foo"
  password             = "******"
  auth_type            = "integrated"
  url                  = "https://my-vcd.com/api"
  org                  = "test"
  vdc                  = "vdc-test"
  allow_unverified_ssl = true
}

data "vcd_org_vdc" "org-vdc" {
  org  = "test"
  name = "vdc-test"
}

data "vcd_provider_vdc" "pvdc" {
  name = data.vcd_org_vdc.org-vdc.provider_vdc_name
}

data "vcd_vm_placement_policy" "policy" {
  name            = "policy1"
  provider_vdc_id = data.vcd_provider_vdc.pvdc.id
}

data "vcd_catalog" "cat" {
  org  = "test"
  name = "cat-test"
}

data "vcd_catalog_vapp_template" "template" {
  name       = "my-template"
  catalog_id = data.vcd_catalog.cat.id
}

# As tenant user with "Organization vDC Compute Policy: View" right
resource "vcd_vm" "vm" {
  provider = vcd.tenant # Using tenant alias

  name             = "foo"
  vapp_template_id = data.vcd_catalog_vapp_template.template.id
  cpus             = 2
  memory           = 2048

  placement_policy_id = data.vcd_vm_placement_policy.policy.id

}
```

Would this work for your use case?

Thanks for the feedback.

**Comment by adambarreiro** [comment]
We were also considering the option to add a new argument to the `vcd_vm_placement_policy` data source that would allow to fetch VM Placement Policies by `vdc_id`, meaning that a tenant user could fetch them without issues if the policy is assigned to the VDC. Would it be helpful for you?

**Comment by adambarreiro** [comment]
> We were also considering the option to add a new argument to the `vcd_vm_placement_policy` data source that would allow to fetch VM Placement Policies by `vdc_id`, meaning that a tenant user could fetch them without issues if the policy is assigned to the VDC. Would it be helpful for you?

We have implemented this option in #948, now a tenant user can use the data source `vcd_vm_placement_policy` as follows:

```hcl
data "vcd_org_vdc" "my-vdc" {
  org  = "test"
  name = "vdc-test"
}

data "vcd_vm_placement_policy" "tf-policy-name" {
  name   = "my-policy"
  vdc_id = data.vcd_org_vdc.my-vdc.id
}
```

It will retrieve the VM Placement Policy "my-policy" if it is assigned to the "vdc-test" VDC.
You can see PR #948 for details.
It would be helpful if you could build the provider from that branch and try it out.

**Comment by NilsBusche** [comment]
Thanks for your ideas and the solution. Unfortunately I have no time at the moment to try out the PR. But looking at your ideas and examples looks fine for me.

**Comment by adambarreiro** [comment]
Closed by #948
